Examples/Examples/All Examples/SceneKitExample.swift

```diff
@@ -0,0 +1,246 @@
+
+        let lightNode = SCNNode()
+        lightNode.light = SCNLight()
+        lightNode.light?.type = SCNLight.LightType.directional
+        lightNode.light?.orthographicScale = 30
+        lightNode.light?.color = UIColor(white: 0.8, alpha: 1.0)
+        lightNode.position = SCNVector3Make(-50, 100, 100)
+        lightNode.light?.zNear = 1
+        lightNode.light?.zFar = 1000
+        lightNode.light?.intensity = 2000
+        lightNode.look(at: modelNode.worldPosition)
+        modelNode.addChildNode(lightNode)
+        let pointNode = SCNNode()
+        pointNode.light = SCNLight()
+        pointNode.light?.type = SCNLight.LightType.omni
+        pointNode.light?.intensity = 3000
+        pointNode.position = SCNVector3Make(0, 25, 0)
+        modelNode.addChildNode(pointNode)
+    }
+
+    internal func makeTranslationMatrix(tx: Double, ty: Double, tz: Double) -> simd_double4x4 {
+        var matrix = matrix_identity_double4x4
+
+        matrix[3, 0] = tx
+        matrix[3, 1] = ty
+        matrix[3, 2] = tz
+
+        return matrix
+    }
+
+    internal func makeScaleMatrix(xScale: Double, yScale: Double, zScale: Double) -> simd_double4x4 {
+        var matrix = matrix_identity_double4x4
+
+        matrix[0, 0] = xScale
+        matrix[1, 1] = yScale
+        matrix[2, 2] = zScale
+
+        return matrix
+    }
+
+    public func render(_ parameters: CustomLayerRenderParameters, mtlCommandBuffer: MTLCommandBuffer, mtlRenderPassDescriptor: MTLRenderPassDescriptor) {
+        let m = parameters.projectionMatrix
+
+        // It is essential to use double precision for computation below: using simd instead
+        // of SceneKit matrix operations.
+        var transformSimd = matrix_identity_double4x4
+        transformSimd[0, 0] = m[0].doubleValue
+        transformSimd[0, 1] = m[1].doubleValue
+        transformSimd[0, 2] = m[2].doubleValue
+        transformSimd[0, 3] = m[3].doubleValue
+        transformSimd[1, 0] = m[4].doubleValue
+        transformSimd[1, 1] = m[5].doubleValue
+        transformSimd[1, 2] = m[6].doubleValue
+        transformSimd[1, 3] = m[7].doubleValue
+        transformSimd[2, 0] = m[8].doubleValue
+        transformSimd[2, 1] = m[9].doubleValue
+        transformSimd[2, 2] = m[10].doubleValue
+        transformSimd[2, 3] = m[11].doubleValue
+        transformSimd[3, 0] = m[12].doubleValue
+        transformSimd[3, 1] = m[13].doubleValue
+        transformSimd[3, 2] = m[14].doubleValue
+        transformSimd[3, 3] = m[15].doubleValue
+
+        // Projection.project(for: modelOrigin, zoomScale: 1.0 / 512.0) corresponds to gl-js's
+        // mapboxgl.MercatorCoordinate.fromLngLat(). origin is in spherical mercator normalized to
+        // 0..1 for the width of the world. In other words, (x,y) E [0..1) is used to represent
+        // coordinates in one world copy, values of x +/- 1 represent wrap.
+        let origin = try! Projection.project(for: modelOrigin, zoomScale: 1.0 / 512.0)
```

Comment:
I still don't understand what the 512 represents here and in the subsequent code; shouldn't the projection calculation here involve `parameters.zoom`?

Reply:
This is a good point. I have apparently approached this from wrong angle; the idea was to complicate matrices and calculation, enhance the complexity with differences between gl-js and gl-native custom layer transformation matrix and Web mercator scaling internals, in order to present 1-1 mapping of gl-js example to approach here. It is wrong and if using parameters.zoom this is much simpler to follow (and less code).